src/lib/apiTypesV2/Registration.cpp

```diff
@@ -71,10 +119,15 @@ std::string DataProvided::toJson(void)
 *
 * Provider::toJson -
 */
-std::string Provider::toJson()
+std::string Provider::toJson(void)
 {
-  JsonHelper jh;
-  // TBD
+  JsonHelper   jh;
+  std::string  urlAsJson = "{\"url\": \"" + url + "\"}";
```

Comment:
This is a bit ugly but ... url is a simple std::string.
I am not very familiar with JsonHelper but I believe we would need to create a new class for `url` to avoid this problem. Not sure it's worth it ...

Reply:
I think the "canonical" solution would be to have a second JsonHelper and do something likes this:

```
JsonHelper jh2;
jh2.addString("url", url);

jh.addRaw("http", jh2.str());
```

The more "styled" solution would be that but, in this particular case, using `urlAsJson` string as it is now is also fine from my point of view. You chose, either "Fixed in" or "NTC" :)

Reply:
Ugly either way. Keeping the initial implementation.

NTC